src/extractor/restriction_map.cpp

```diff
@@ -143,11 +143,13 @@ bool RestrictionMap::CheckIfTurnIsRestricted(const NodeID node_u,
         {
             return true;
         }
-        if (node_w != restriction_target.target_node && // target not found
-            restriction_target.is_only)                 // and is an only restriction
-        {
-            return true;
-        }
+        // The following check is actually perfomed in intersection generation where we can also
+        // verify if the only restriction is valid at all.
+        // if (node_w != restriction_target.target_node && // target not found
+        //    restriction_target.is_only)                 // and is an only restriction
+        // {
+        //    return true;
+        // }
```

Comment:
Well I thought about this. I feel this comment is a good documentation to nod add this check again in the future. But we can also remove it.